homework-assignments-1/task-3.ts

```diff
@@ -0,0 +1,24 @@
+class Book {
+    title: string;
+    author: string;
+    isbn: string;
+    
+    constructor (title: string, author: string, isbn: string) {
+        this.title = title;
+        this.author = author;
+        this.isbn = isbn;
+    }
+}
+
+class Library {
+    books: Book[] = [];
+
+    addBook(book: Book): void {
+        this.books.push(book);
+    }
+
+    removeBook(isbn: string) {
+        return this.books.filter((book) => book.isbn !== isbn);
```

Comment:
This returns newly updated array but doesn't mutate original array. You should do it using slice 
` removeBook(isbn: string): void {
    const index = this.books.findIndex(book => book.isbn === isbn);
    if (index !== -1) {
      const removedBook = this.books.splice(index, 1);
      console.log(`Removed book: ${removedBook[0].title}`);
    } else {
      console.log(`Book with ISBN ${isbn} not found.`);
    }
  }`